src/platform/common/crypto.ts

```diff
@@ -14,7 +14,7 @@ import * as hashjs from 'hash.js';
  */
 @injectable()
 export class CryptoUtils implements ICryptoUtils {
-    public createHash(data: string, algorithm: 'SHA512' | 'SHA256' = 'SHA256'): string {
+    public async createHash(data: string, algorithm: 'SHA512' | 'SHA256' = 'SHA256'): Promise<string> {
```

Comment:
We're just pretending it's async? Why would we want to do this?

Reply:
next pr will add the real bits, wanted to break this work into smaller chunks. one pr already touched almost 70 files

Reply:
i merged another pr into this, this pr now has 74 files changed.
hopefully it makes sense why im trophy to break the work into multiple prs.

again, another pr will add the others changes, adding that here will only make this much larger and more difficult to review